From 48fa303cb6820f5afff4f7c0b2db3f5fc4b1139b Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Fri, 23 Jun 2017 02:37:17 +0300
Subject: [PATCH 2/2] Add library versioning support

This adds library soname versioning support. Now cmake
builds libclblast.so.<major>.<minor>.<patch> binary,
and creates libclblast.so.<major> and libclblast.so
symbolic links instead of just building libclblast.so.
This helps to avoid QA errors with Yocto.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c602c5e..609fa68 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -204,6 +204,10 @@ else(BUILD_SHARED_LIBS)
   add_library(clblast STATIC ${SOURCES})
 endif()
 
+set_target_properties(clblast PROPERTIES
+                      VERSION ${clblast_VERSION_MAJOR}.${clblast_VERSION_MINOR}.${clblast_VERSION_PATCH}
+                      SOVERSION ${clblast_VERSION_MAJOR} )
+
 target_link_libraries(clblast ${OPENCL_LIBRARIES})
 
 # Includes directories: CLBlast and OpenCL
-- 
2.7.5

